site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+
+The vr test code demonstrates how to use this API:
+
+	- `test_vr.c <https://github.com/kripken/emscripten/blob/master/tests/test_vr.c>`_
+
+.. _functions-vr-api:
+
+Functions
+---------
+
+Initialization
+==============
+
+.. c:function:: int emscripten_vr_init()
+
+	Initialize the emscripten VR API. This will `navigator.getVRDisplays()
+	<https://w3c.github.io/webvr/spec/1.1/#navigator-getvrdisplays-attribute>`_
+	and when completed, set the return of :c:func:`emscripten_vr_ready` to be `true`.
+
+	:returns: `1` on success, `0` if the browsers WebVR support is insufficient.
+	:rtype: int
+
+.. tip:: This call succeeding is not sufficient for use of the rest of the API. Please
+	make sure to wait until the return value of :c:func:`emscripten_vr_ready` is true.
+
+.. c:function:: int emscripten_vr_ready()
+
+	Check whether the VR API has finished initializing VR displays.
+	See also :c:func:`emscripten_vr_init`.
+
+	This function may return `1` event if WebVR is not supported in the
+	running browser.
+
+	:returns: `1` if ready, `0` otherwise.
+	:rtype: int
+
+API Queries
+===========
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called but do not require VR displays and can therefore be called before the return
+value of :c:func:`emscripten_vr_ready` is `true`.
+
+.. c:function:: int emscripten_vr_version_minor()
+
+	Minor version of the WebVR API currently supported by the browser.
+
+	:returns: minor version of WebVR, or `-1` if not supported or API not initialized.
+	:rtype: int
+
+.. c:function:: int emscripten_vr_version_major()
+
+	Major version of the WebVR API currently supported by the browser.
+
+	:returns: major version of WebVR, or `-1` if not supported or API not initialized.
+	:rtype: int
+
+Display Functions
+=================
+
+All of the following functions require :c:func:`emscripten_vr_init` to have been
+called the return value of :c:func:`emscripten_vr_ready` to be `true`.
+
+.. c:function:: int emscripten_vr_count_displays()
+
+	:returns: Number of displays connected.
+	:rtype: int
+
+.. c:function:: VRDisplayHandle emscripten_vr_get_display_handle(int displayIndex)
+
+	:param int displayIndex: index of display (inclusive 0 to exclusive :c:func:`emscripten_vr_count_displays`).
+	:returns: handle for a VR display.
+	:rtype: VRDisplayHandle
+
+.. c:function:: char* emscripten_vr_get_display_name(VRDisplayHandle handle)
+
+	Get a user-readable name which identifies the VR display.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: name of the VR display or `0 (NULL)` if the handle is invalid.
+	:rtype: char*
+
+.. c:function:: bool emscripten_vr_get_display_connected(VRDisplayHandle handle)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: `true` if the display is connected, `false` otherwise or when
+		the handle is invalid.
+	:rtype: bool
+
+.. c:function:: bool emscripten_vr_get_display_presenting(VRDisplayHandle handle)
```

Comment:
Also here `emscripten_vr_get_display_presenting` vs `emscripten_vr_display_presenting`.

Reply:
Haven't jumped in to read the spec in detail on these nuances, though here I wonder if apps are expected to be polling these functions (e.g. in the beginning of each render frame?) for whether they are still presenting and connected, or are there notification events involved for disconnection and "lost present"? (or for the same way, "connected" and "acquired present"?)

Reply:
There are events for this, but these two functions are implemented using a property of `VRDisplay`.

Most examples seem to check `display.isPresenting` every frame (e.g. [display present example from webvr.info](https://github.com/toji/webvr.info/blob/master/samples/03-vr-presentation.html#L204)).

As noted in [this comment](https://github.com/kripken/emscripten/pull/5508#discussion_r143470069), WebVR specific events are not implemented with this pullrequest currently.